src/main.ts

```diff
@@ -6,51 +6,115 @@
 
 // based on https://github.com/Microsoft/vscode/commit/41f0ff15d7327da30fdae73aa04ca570ce34fa0a
 
-import { ExtensionContext, workspace, window, Disposable, commands, Uri, OutputChannel } from 'vscode';
+import { ExtensionContext, workspace, window, Disposable, commands, Uri, OutputChannel, WorkspaceFolder } from 'vscode';
 import { HgFinder, Hg, IHg, HgFindAttemptLogger } from './hg';
 import { Model } from './model';
 import { CommandCenter } from './commands';
 import { HgContentProvider } from './contentProvider';
 import * as nls from 'vscode-nls';
+import * as path from 'path';
+import * as fs from 'fs';
 import typedConfig from './config';
 
 const localize = nls.config(process.env.VSCODE_NLS_CONFIG)();
 
+async function isHgRepository(folder: WorkspaceFolder): Promise<boolean> {
+	if (folder.uri.scheme !== 'file') {
+		return false;
+	}
+
+	const dotHg = path.join(folder.uri.fsPath, '.hg');
+
+	try {
+		const dotHgStat = await new Promise<fs.Stats>((c, e) => fs.stat(dotHg, (err, stat) => err ? e(err) : c(stat)));
+		return dotHgStat.isDirectory();
+	} catch (err) {
+		return false;
+	}
+}
+
+async function warnAboutMissingHg(): Promise<void> {
```

Comment:
This function should be in `interaction.ts`, which is where this extension keeps stuff related to UI. This is different from how the Git extension does it.